bin/detect

```diff
@@ -23,18 +23,22 @@
 BP=$(dirname "$(dirname "$0")")
 
 # Generally stacks do not have ruby or python installed, with the exception of cflinuxfs3, so we install them here
-# We deliberately skip re-installing python and ruby on the cflinuxfs3 stack to avoid compatibility issues
-# (e.g. the bootstrapped python is incompatible with cflinuxfs3 due to being compiled on a newer version of GBLIC)
+#
+# We skip re-installing ruby on the cflinuxfs3 stack to avoid compatibility issues
 if [ "$CF_STACK" != "cflinuxfs3" ]; then
-  PYTHON_DIR="/tmp/php-buildpack/python"
-  mkdir -p "${PYTHON_DIR}"
-  source "$BP/bin/install-python" "$PYTHON_DIR" "$BP" &> /dev/null
-
   RUBY_DIR="/tmp/php-buildpack/ruby"
   mkdir -p "${RUBY_DIR}"
   source "$BP/bin/install-ruby" "$RUBY_DIR" "$BP" &> /dev/null
 fi
 
+# To avoid having to support both python 2 & 3 and to avoid using the ancient
+# python included in cflinuxfs3, always install python, unless running unit tests
+if [ -z "${USE_SYSTEM_PYTHON}" ]; then
```

Comment:
I'm not super happy about having this conditional here, but the unit tests run these scripts and were breaking when installing another python.

I'd be interested in exploring other options so that we can always, always install pythons.